File: AssignmentEcommerce_Backend/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AssignmentEcommerce_Backend.Models
{
    public class Product
    {
        public string ProductId { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public decimal Price { get; set; }

        public string Images { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime UpdatedDate { get; set; }

        public string CategoryId { get; set; }

        public Category Category { get; set; }
    }
}

File: angular-ec/src/models/product.ts
export class Product {
  productId: string;
  name: string;
  description: string;
  price: number;
  images: string;
  categoryId: string;
  nameCategory: string;
  rating: number;

  constructor(
    productId: string,
    name: string,
    description: string,
    price: number,
    images: string,
    categoryId: string,
    nameCategory: string,
    rating: number
  ) {
    this.productId = productId;
    this.name = name;
    this.description = description;
    this.price = price;
    this.images = images;
    this.categoryId = categoryId;
    this.nameCategory = nameCategory;
    this.rating = rating;
  }
}
